gateway/gateway.go

```diff
@@ -283,7 +283,10 @@ func (g *gatewayImpl) open(ctx context.Context) error {
 
 	var readyOnce sync.Once
 	readyChan := make(chan error)
-	go g.listen(t, func(err error) {
+
+	messagesChan := make(chan event, g.config.MessageBufferSize)
+	go g.dispatcher(messagesChan)
```

Comment:
NIT: Would move this above the var readyOnce to keep it together with g.listen, where its used